x-pack/plugins/actions/server/routes/get.test.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getActionRoute } from './get';
+import { mockRouter, RouterMock } from '../../../../../src/core/server/http/router/router.mock';
+import { mockLicenseState } from '../lib/license_state.mock';
+import { verifyApiAccess } from '../lib/license_api_access';
+import { mockHandlerArguments } from './_mock_handler_arguments';
+
+jest.mock('../lib/license_api_access.ts', () => ({
+  verifyApiAccess: jest.fn(),
+}));
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('getActionRoute', () => {
+  it('gets an action with proper parameters', async () => {
+    const licenseState = mockLicenseState();
+    const router: RouterMock = mockRouter.create();
+
+    getActionRoute(router, licenseState);
+
+    const [config, handler] = router.get.mock.calls[0];
+
+    expect(config.path).toMatchInlineSnapshot(`"/api/action/{id}"`);
+    expect(config.options).toMatchInlineSnapshot(`
+      Object {
+        "tags": Array [
+          "access:actions-read",
+        ],
+      }
+    `);
+
+    const getResult = {
+      id: '1',
+      actionTypeId: '2',
+      name: 'action name',
+      config: {},
+    };
+    const actionsClient = {
+      get: jest.fn().mockResolvedValueOnce(getResult),
+    };
+
+    const [context, req, res] = mockHandlerArguments(
+      { actionsClient },
+      {
+        params: { id: '1' },
+      },
+      ['ok']
+    );
+
+    expect(await handler(context, req, res)).toMatchInlineSnapshot(`
+      Object {
+        "body": Object {
+          "actionTypeId": "2",
+          "config": Object {},
+          "id": "1",
+          "name": "action name",
+        },
+      }
+    `);
+
+    expect(verifyApiAccess).toHaveBeenCalledWith(licenseState);
```

Comment:
I don't the test below this one - "ensures the license allows getting actions" is required, since the thing it's testing is this particular `expect()`.

Reply:
I prefer the explicit test and removing the redundant expect -test are just as much about documentation as about logical testing.

Reply:
generally speaking, that's true.  But it adds 30 lines to 5 test cases, so 150 more lines of tests that are technically redundant.  I can go either way.  Some day we'll spend some time and framework-ize some of this test boilerplate, will be less of an issue to me :-)